bfrec

```diff
@@ -153,7 +153,7 @@ uefi_capsule_update()
 
     # Set capsule update file variable name
     capsule_efi=MmcBootCap
-    device_efi=/dev/mmcblk0p1
+    device_efi=$(lsblk -lp | grep -i efi | cut -d' ' -f1)
```

Comment:
Might need to check two things
1. Whether the 'grep -i efi' works for all existing Linux distribution. What is efi partition is not mounted yet?
Or maybe "fdisk -l | grep -i EFI | cut -d' ' -f1" would be more reliable since it doesn't rely on mount point?
2. What if there are EFI partitions on eMMC as well? I remember I saw such use cases before.  Probably we could pick the first one (at least no failure in such case)?

Reply:
Both the eMMC and SSD can have an ESP.  Depends on the BOOT options which one would be used.

Reply:
> I think an ESP partition is identified by the partition standard UUID (C12A7328-F81F-11D2-BA4B-00A0C93EC93B). Maybe we should use that to ID it? Not to be confused with the FS UUID.

I think it's the same as the 'fdisk -l" output, see below.
===============================================
root@OPT166-Development-bf:~# fdisk -l
Disk /dev/mmcblk0: 14.26 GiB, 15300820992 bytes, 29884416 sectors
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
Disklabel type: gpt
Disk identifier: A2DF9E70-6329-4679-9C1F-1DAF38AE25AE

Device          Start      End  Sectors  Size Type
/dev/mmcblk0p1   2048   104447   102400   50M EFI System
/dev/mmcblk0p2 104448 29884382 29779935 14.2G Linux filesystem
===============================================

When doing "fdisk /dev/mmcblk0", we can use "l" command to list all the TYPEs, where "EFI System" is identified by UUID  C12A7328-F81F-11D2-BA4B-00A0C93EC93B

Command (m for help): l
  1 EFI System                     C12A7328-F81F-11D2-BA4B-00A0C93EC93B

Reply:
Could you please help review whenever you have a chance? Thanks,